migrations/artifact_tenant_token_migration.py

```diff
@@ -0,0 +1,87 @@
+import boto3
+import sys
+import pymongo
+import os
+
+l.basicConfig(level=l.DEBUG)
+s3 = boto3.resource('s3')
+
+
+class Migration(object):
+
+    mc = None
+    s3_client = None
+    s3_bucket = None
+    tenant_dbs = []
+
+    def __init__(self, mc, s3_client, s3_bucket):
+        self.mc = mc
+        self.s3_client = s3_client
+        self.s3_bucket = s3_bucket
+
+        # make sure we are properly connected to the mongodb server
+        assert self.mc.server_info() is not None
+    
+        # make sure we actually can access the bucket passed
+        assert self.s3_bucket in s3.buckets.all(), "failed to find %s bucket" % self.s3_bucket
+
+    def set_s3_tenant_tag(self, obj_key, tenant):
+        new_tag = {"TagSet" :[{"Key": "tenant_id", "Value": tenant}]}
+
+        l.info("setting tags [%s] for: %s" % (new_tag, obj_key))    
+        self.s3_client.put_object_tagging(Bucket=self.s3_bucket.name, Key=obj_key, Tagging=new_tag)
+
+
+    def copy_to_new_obj(self, old_key, new_key):
+        l.info("copying object: %s to %s" % (old_key, new_key))
+        self.s3_client.copy_object(Bucket=self.s3_bucket.name, CopySource="%s/%s" % (self.s3_bucket.name, old_key), Key=new_key)
+        self.s3_client.delete_object(Bucket=self.s3_bucket.name, Key=old_key)
+
+    def set_tenant_databases(self):
+        tenant_dbs = []
+
+        for db in self.mc.database_names():
+            if db.startswith("deployment_service-"):
+                if len(db.split("-")) != 2:
+                    raise SystemExit("unable to parse tenant ID from database name: %s" % db)
+                tenant_dbs.append(db)
+            else:
+                l.debug("skipping over : %s, as it doesn't look like a deployment_service_<tenant> database" % db)
+
+        self.tenant_dbs = tenant_dbs
+
+    def perform(self):
+        for db in self.tenant_dbs:
+            tenant_token = db.split("-")[1]
+
+            l.info("tenant token extract from database: %s" % (tenant_token))
+
+            d =self.mc.get_database(db)
+            cursor = d.get_collection("images")
+
+            for d in cursor.find({}):
+                try:
+                    image_id = d['_id']
+                except IndexError:
+                    raise SystemExit("failed to find '_id' in image, this should never happen!")
+        
+                # set the tenant_id tag to the object
+                self.set_s3_tenant_tag(image_id, tenant_token)
+                new_object_key = "%s/%s" % (tenant_token, image_id)
+
+                l.info("%s has an artifact: %s that should be renamed to: %s" % (db, image_id, new_object_key)) 
+                self.copy_to_new_obj(image_id, new_object_key)
+
+
+if __name__ == "__main__":
+    MC = pymongo.MongoClient(os.getenv("MONGODB_MENDER"))
+    BUCKET = "mender-artifacts-greg"
+    S3_CLIENT = boto3.client('s3')
+    S3_BUCKET = s3.Bucket(BUCKET)
+    
+    migration = Migration(MC, S3_CLIENT, S3_BUCKET)
+    migration.set_tenant_databases()
```

Comment:
just a general note but not a blocker.

instead of making the Migration stateful, what about an alternative pattern?
- static method get_tenant_databases(), returns list
- `perform(databases)`

then you don't have to remember the `.set_tenant_databases`, without which the instance is unusable and it's easy to miss. stateful classes of course make sense, but not in this case imo.

with passing tenant dbs - the interface of the class is simple, you can't misuse it. it doesn't make much difference in a one-off tool, but in general I'd prefer it for the sake of clarity.

personally what I would do however is: embed the tenant db iteration within `perform()`. then it's even simpler. wrap it up in a helper for clarity but keep it private and don't access explicitly.